sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -120,6 +120,17 @@ object ExternalCatalogUtils {
       new Path(totalPath, nextPartPath)
     }
   }
+
+  def getExtraPartPathCreatedByHive(
```

Comment:
A general suggestion. When the function names become not self-descriptive. Write the function comments. 

For example, here, please add comments for `generatePartitionPath` and `getExtraPartPathCreatedByHive`